source/agora/common/Task.d

```diff
@@ -51,4 +51,26 @@ public class TaskManager
         static import vibe.core.core;
         vibe.core.core.sleep(dur);
     }
+
+    /***************************************************************************
+
+        Run an asynchronous task after the set time in vibe.d's
+        can set it to repeat or run it only once.
+
+        Params:
+            timeout = Determines the minimum amount of time that elapses before
+                the timer fires.
+            callback = If non-null, this delegate will be called when the timer
+                fires
+            periodic = Speficies if the timer fires repeatedly or only once
+
+    ***************************************************************************/
+
+    public void timer (Duration timeout, scope void delegate() dg,
+        bool periodic = true)
+    {
+        import vibe.core.core;
+        auto tm = setTimer(timeout, dg, periodic);
+        scope (exit) tm.stop();
```

Comment:
@TrustHenry could you explain the logic to me here? You are setting a timer in the statement before, and then you are immediately stoping it.

Reply:
Also having a `scope (exit)` at the end of a function does not make any sense, it is the same thing as writing the statement without the `scope (exit)`.

Reply:
You're right.